.github/workflows/ci.yml

```diff
@@ -133,10 +133,21 @@ jobs:
         shell: bash
         run: echo ::set-output name=flavor::$([[ "${{ matrix.os }}" == "cuda" ]] && echo "Cuda" || echo "CPU")
 
+      # Clean tests (kernel caching disabled) use more CPU and memory, so we only run them for
+      # scheduled runs, pushes to master, and pushes to tags.
+      # Parallelization also needs to be disabled to avoid Out of Memory errors.
+      - name: Set environment variables
+        id: env
+        shell: bash
+        run: |
+          echo "::set-output name=json::{}"
+          if [[ ${{ github.event_name == 'schedule' || (github.event_name == 'push' && (github.ref == 'refs/heads/master' || startsWith(github.ref, 'refs/tags/v'))) }} ]]; then
+            echo "::set-output name=json::{\"ILGPU_CLEAN_TESTS\":\"true\", \"ILGPU_TEST_PARALLELIZATION_LEVEL\":\"1\"}"
+          fi
```

Comment:
one last nitpick: maybe we should set a `json` variable on lines 143 and 145, and then `echo` it after the condition?
the current code will `echo` twice when the condition is met, which GitHub Actions seems to be happy about, but it doesn't _feel_ ideal.

Reply:
@ljubon do you think you can refactor this piece of code and create a final commit? i think we did similar thing elsewhere as well

Reply:
I'm on it, currently verifying...once it's done, I'm pushing here ⚡

Reply:
you can find example workflow here (tested, it works) pushing in min ⚡